### ISSUE TITLE
Password with "*" in in will not be censored in Raw Content in Profiler 

### ISSUE THREAD

**Issue by ExtremTechniker** [symfony/symfony#53603]
### Symfony version(s) affected

6.3.9,6.4.2,...

### Description

Inputted Password e.g. in an Loginform will be censored with asterisk in the Profiler. If password input contains an asterisk the password will be censored in the `POST` view but not in the raw content view.

### How to reproduce

- Login Form (no need for special)
- Input Password with asterisk `*`
- Go to Request in Profiler
- Look at Post Body and Raw Content

### Possible Solution

Censor the password in the raw content like in the post body

**Comment by teamflp** [comment]
- **Identify the Concerned Component:** Locate the component in the Symfony Profiler that is responsible for displaying the request data, particularly the raw content view.

- **Modify the Censorship Logic:**
Modify the code to apply the same password censoring logic, which is already in place for the POST view, to the raw content view. This might involve analyzing the content of the request data and replacing the values of password fields with asterisks or another placeholder.

**Comment by xabbuh** [comment]
I cannot reproduce the issue. With the Symfony Demo it looks like in the following two screenshots.

![Bildschirmfoto 2024-01-23 um 13 07 06](https://github.com/symfony/symfony/assets/1957048/e056171c-e301-4c9a-9217-e32c55447e41)

![Bildschirmfoto 2024-01-23 um 13 07 28](https://github.com/symfony/symfony/assets/1957048/13066173-0ef8-4459-935e-0be828108288)

Can you create a small example application that allows to reproduce your issue?

**Comment by xabbuh** [comment]
I am going to close here for now due to the lack of feedback. Please let us know when you have more information and we can consider to reopen.

**Comment by ExtremTechniker** [comment]
Hi,

sorry for the lack of feedback. I just looked deeper into this issue. Now I can clearly say that there is an issue.  I've also tested with the Symfony Demo.

The error come from the `\Symfony\Component\HttpKernel\DataCollector\RequestDataCollector` in here there the password will be obfuscated. https://github.com/symfony/http-kernel/blob/354bedd6b2996b5a1f3520e4b4e8efc38db071c9/DataCollector/RequestDataCollector.php#L125-L129

`$-_.+!*'(),?` are safe characters that must not be encoded. But rawurlencode encodes the `*` to `%2A` afterwards the `str_replace` can't find the password. 

Possible solution convert back `%2A` into `*` after `rawurlencode`.
